### DISCUSSION TITLE
the longstanding "Doodles persist" issue - is it browser-dependent?

### DISCUSSION THREAD

**Discussion by ddbuscombe** [dbuscombe-usgs/dash_doodler#49] [General]
Doodler uses a timer to periodically search through the `assets` and `labeled` folders, reconcile them, and update the image list. This is to ensure that you dont doodle the same image twice

The code that does that is https://github.com/Doodleverse/dash_doodler/blob/main/app.py#L482

(this is why you often have to select images a few times before they 'stick')

That is also supposed to clear the doodles from the main window. Often, it does not; doodles 'persist' from the previous image. The only known workaround is a browser refresh, which is annoying (-ly slow)

Changing `interval=500` to a higher number, say `interval=1000`, or a lower number, say `interval=100`, did not help

Switching browsers, from Chrome to Firefox, helped a LOT. Hey The 4 MuskaDoodlers! @venuswku @ebgoldstein @2320sharon @CameronBodine - you seeing the same thing? Which browser do you use? have you noticed big differences between them?

**Comment by ebgoldstein** [comment]
I use firefox. file selection is rarely a problem. I typically wait ~2 seconds after selecting an image before i slip over to the doodling interface. 

(is the same selection issue as #12 ?)

**Comment by 2320sharon** [comment]
I typically use Google Chrome when I'm doodling. I recall having the persisting doodles error occur occasionally, but I have encountered it since the latest update to doodler. That being said I also haven't been doodling much.

**Comment by 2320sharon** [comment]
![doodler_file_selection_bug_chrome_gif](https://user-images.githubusercontent.com/61564689/199579181-e0920427-bf72-466d-8879-24aaaeb06e61.gif)
Also @dbuscombe-usgs not sure if you're aware of this quirk in doodler, but on Google Chrome when the user initially loads up doodler none of the files are shown. They have to switch back to the main page of doodler and then click the file tab again for the files to appear. This has happened every single time I have opened doodler with Google Chrome.

**Comment by ddbuscombe** [comment]
Thanks @2320sharon that's good input. I noticed only yesterday how bad Chrome was w/r/t the 'persistent doodle' problem. It almost went away switching to Firefox. It seems like everyone has a slightly different experience ... it would be nice to understand what exactly happens when those doodles persist. I don't know enough about browsers and caching to really make headway. But thanks for the info!!

**Comment by 2320sharon** [comment]
Happy to help! For now, it might be a good idea to recommend that users use FireFox. Digging into browser-specific issues can be a pain and sometimes result from the package you're using to compile python code to javascript rather than your code.

**Comment by venuswku** [comment]
I have a similar but slightly different experience as Sharon on Google Chrome (doodles persist and no files are shown in the dropdown). But instead of nothing appearing in the dropdown initially, my files initially appear twice:
![image](https://user-images.githubusercontent.com/71415129/199774210-59e4b091-8f1f-4b37-ae61-ba560cdb25de.png)
I don't see my files after doodling and segmenting an image but refreshing displays the other unsegmented files once like normal.
Another tiny thing I noticed was when I accidentally clicked anywhere on the "Doodler: Interactive Image Segmentation" bar at the top, it opens a window for saving something:
![image](https://user-images.githubusercontent.com/71415129/199777628-46070358-5434-469c-94e3-cbdbeb96ce5f.png)
I never tried saving anything though so I'm not sure what gets saved.

**Comment by CameronBodine** [comment]
I use Firefox, and generally don't experience the file or doodle sticking issue. It does happen occasionally, but I refresh the browser and generally that works. I have not updated Doodler with your most recent changes, however.

**Comment by CameronBodine** [comment]
I have been doodling more lately, and have a bit more feedback. Doodles seem to persist more often when you don't deselect `compute/show segmentation` before switching to `File List and Instructions` tab to select a new image (see #54 for more info). If the doodle does stick, my workaround is to repeatedly select _different_ images in the `Select Image` drop-down menu. I know that the doodles have cleared when my `User ID: "csb"` disappears for a second then reappears. At that point, I can select the image I want and switch back to the `Imagery and Controls` tab to start doodling the image. Hope this helps others!

**Comment by ddbuscombe** [comment]
Thanks for the input @CameronBodine this is helpful
